### ISSUE TITLE
_ -> nothing should be a named function

### ISSUE THREAD

**Issue by Cvikli** [FluxML/Zygote.jl#903]
Hey I was reading this:
`@adjoint ignore(f) = ignore(f), _ -> nothing`
in the utils.jl:26 

The function name: "#34#35" which isn't too nice in many precompilation case. Can we ask some:
`noop(_) = nothing` to by used instead?
Or are there any better way?

**Comment by willtebbutt** [comment]
Good point. This should be called `ignore_pullback` as in "pullback of ignore" if we're following the `ChainRules` convention. I would be very happy to review a PR that fixes this.

**Comment by DhairyaLGandhi** [comment]
It's used in a lot of places in zygote, but I agree it would be nice.

Otoh, we might want to distinguish between different places this shows up for better stacktraces, and seems worth it

**Comment by willtebbutt** [comment]
> Otoh, we might want to distinguish between different places this shows up for better stacktraces, and seems worth it

I'm not actually sure whether this an argument for or against the proposed change. Is there some advantage to the current implementation?

**Comment by DhairyaLGandhi** [comment]
Im pro this change! As you mentioned the ignore_pullback, I'm taking it a step forward saying we can name other places this shows up in more descriptively as well. I'm not sure if it's worth the code smell since I'm not sure it would actually help traces but I'd definitely want to see the difference

**Comment by willtebbutt** [comment]
Ahh I see. In my experience (IIRC) having these names in the traces is a _big_ help, both when trying to figure out where an error is coming from and when digging around in `@code_warntype` etc.
